### PR TITLE
Locality testing

### PR DESCRIPTION
**Pull request by bruskiza**
- For SimonC to investigate.

### CODE REVIEW

**Comment by hodgestar** [comment]
Travis builds are failing because `underscore` isn't listed in `package.json`.

File: src/app.js [hunk 1]
```diff
@@ -1,12 +1,38 @@
...
+            var provinces = self.im.sandbox_config.get("locations.json", {json: true});
+            provinces = JSON.parse(fs.readFileSync("src/lookups/provinces.json", "utf8"));
+            console.log("app.js");
+            console.log(provinces);
+            console.log("/app.js");
+            
+            province_array = [];
+            _.each(provinces, function(val) {
+                province_array.push(val.Province);
+            });
```
Comment (hodgestar) [inline src/app.js:87]: This should be something more like:

``` javascript
var choices = _.map(provinces, function(obj) {
    return new Choice(obj.Province, obj.Provice);
});
```

See http://vumi-jssandbox-toolkit.readthedocs.org/en/master/states/reference.html#Choice.

Comment (hodgestar) [inline src/app.js:87]: Please also read http://vumi-jssandbox-toolkit.readthedocs.org/en/master/states/reference.html#PaginatedChoiceState more carefully. There are two options -- `options_per_page` and `characters_per_page`. You can either limit options per page or characters per page. You probably want characters per page. That requires `{options_per_page: null, characters_per_page: 160}`.

P.S. Sorry for forgetting about this rough edge.

File: go-app.js [hunk 2]
```diff
@@ -8,12 +8,39 @@ go;
...
+            console.log("app.js");
+            console.log(provinces);
+            console.log("/app.js");
+
+            var choices = _.map(provinces, function(obj) {
+                return new Choice(obj.Province, obj.Provice);
+            });
+
+            return new PaginatedChoiceState(name, {
+                question: 'What province do you belong to?' + provinces,
```
Comment (hodgestar) [inline go-app.js]: Ah! You need to remove the `+ provinces` here. Doh. :)

**Comment by smn** [comment]
Closing because at this point it makes more sense to start a new PR to do this rather than deal with all the merge conflicts from this early attempt.

**Pull request closed without merging**